consensus/istanbul/backend/handler.go

```diff
@@ -286,3 +301,172 @@ func (sb *Backend) UnregisterPeer(peer consensus.Peer, isProxiedPeer bool) {
 		}
 	}
 }
+
+// Handshake allows the initiating peer to identify itself as a validator
+func (sb *Backend) Handshake(peer consensus.Peer) (bool, error) {
+	// Only written to if there was a non-nil error when sending or receiving
+	errCh := make(chan error)
+	isValidatorCh := make(chan bool)
+
+	sendHandshake := func() {
+		var validatorProofMsg *istanbul.Message
+		var err error
+		peerIsValidator := peer.PurposeIsSet(p2p.ValidatorPurpose)
+		if peerIsValidator {
```

Comment:
I was thinking about this, and it seems to be that non-proxy full-nodes would also create and send their own self-signed enode certificate to a peer if it is marked as a validator, even though the full-node, if operating correctly would not want to be considered a validator by that peer. (In practice, this would not happen under most circumstances because they should not have any ValidatorPurpose connections)
Given this, you might add an `sb.coreStarted` check here as a way to see if this node intends to be seen as a validator.

Reply:
Checking `sb.coreStarted` wouldn't work if this node is a proxy that wants to be seen as its proxied validator. If the node isn't using modified code, I don't see a situation where a non-proxy full node would have an incorrect ValidatorPurpose peer

Reply:
Oh right 🙈Looks good to me